exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1695,23 +1747,23 @@ func (b *Bitfinex) subscribeToChan(chans []subscription.Subscription) error {
 	// Add a temporary Key so we can find this Sub when we get the resp without delay or context switch
 	// Otherwise we might drop the first messages after the subscribed resp
 	c.Key = subID // Note subID string type avoids conflicts with later chanID key
-
-	c.State = subscription.SubscribingState
-	err = b.Websocket.AddSubscription(&c)
+	if err = c.SetState(subscription.SubscribingState); err == nil {
+		err = b.Websocket.AddSubscriptions(c)
+	}
 	if err != nil {
-		return fmt.Errorf("%w Channel: %s Pair: %s Error: %w", stream.ErrSubscriptionFailure, c.Channel, c.Pair, err)
+		return fmt.Errorf("%w Channel: %s Pair: %s Error: %w", stream.ErrSubscriptionFailure, c.Channel, c.Pairs, err)
 	}
 
 	// Always remove the temporary subscription keyed by subID
-	defer b.Websocket.RemoveSubscriptions(c)
+	defer b.Websocket.RemoveSubscriptions(c) //nolint:errcheck // Ignore error, since we couldn't do anything with it
```

Comment:
Understood.
But it also means more allocs, both for the func, lexical stack, etc.
I know it's not important, but I really _loathe_ paying runtime costs to placate the linter

Fixed 1b228453

Reply:
Oh interesting, I did not realise. I may go back on that depending on impact

Reply:
It's marginal. But that was my reasoning, anyway.
I also woke up regretting removing the comment, because an explicit explanation is still justified.

I land on the "leave it as it was" side of the fence right now, personally.
Your call, captain.